### ISSUE TITLE
Regexp or other advanced syntax in Notifications Keywords feature and short explanation how to use them

### ISSUE THREAD

**Issue by MurzNN** [vector-im/element-meta#1872]
#### What would you like to do?
Now we can only use plain text to set custom keywords in Notifications settings. Will be good to add ability to use regex or other advanced syntax like `first part*last part` or `hello from:@alice`, `attention alias:*mycompany.org` to make keywords more customizable.

Also it's unclear is it case sensitive, or not? Can we use HTML tags in it, or it analyses only plain text (`body` field instead of `formatted_body`) version of message? Can we put sender mxid to it (to be notified of any message from specific user, eg `@bot:fosdem.org`) and it should work if all JSON of event is matched, or only message content is analysed? Can we use part of room alias or room name to filter configured keywords by rooms/spaces?

So as first step will be good to explain those info in short description text in Element interface.

#### Why would you like to do it?

For example, in FOSDEM rooms I want to be noisy notified about each new session start. So I can use the `Up next` part of phrase, but it will have false positives in other rooms and from other messages.

#### How would you like to achieve it?
Allow regexp expressions in Keywords.

**Comment by MurzNN** [comment]
Can't find anything about Notifications Keywords in Matrix Spec https://spec.matrix.org/v1.2/ - is this feature described somewhere in it?

**Comment by robintown** [comment]
@MurzNN They probably correspond to [`event_match` push rules](https://spec.matrix.org/v1.2/client-server-api/#conditions-1)

**Comment by MurzNN** [comment]
Yeah, push rules are designed very flexible! So seems we can implement more complex keywords notifications without changing the spec, that's good! But sadly it's without support of regexp now :(

**Comment by t3chguy** [comment]
There's no special UI needed here, the keyword you enter in the UI is treated as a glob by the server as per the docs.

> pattern: The glob-style pattern to match against. Patterns with no special glob characters should be treated as having asterisks prepended and appended when testing the condition.

e.g `*first part*last part*` should work

**Comment by MurzNN** [comment]
So maybe simply explain this as description for that field in UI? Together with description in which exact event's fields this matching will happen.

**Comment by t3chguy** [comment]
Yup, probably
